synapse/python_dependencies.py

```diff
@@ -139,12 +142,12 @@ def dependencies(self):
             yield "'" + i + "'"
 
 
-def check_requirements(for_feature=None):
+def check_requirements(for_feature=None, dev=False):
     deps_needed = []
     errors = []
 
     if for_feature:
-        reqs = CONDITIONAL_REQUIREMENTS[for_feature]
+        reqs = CONDITIONAL_REQUIREMENTS.get(for_feature, [])
```

Comment:
Why this change? It seems unrelated?

Reply:
if `for_feature` is not a key in `CONDITIONAL_REQUIREMENTS`, it'll raise a `KeyError`, i think the right logic then is to return a default empty list

This is so that when `check_requirements` is called with a dependency list that does not exist anymore, any time in the future, it should not return a cryptic error, should i revert this?